schema/agent/1.0.0/schema.json

```diff
@@ -0,0 +1,224 @@
+      },
+      "tools": {
+        "type": "array",
+        "description": "List of tools utilized by the agent",
+        "items": {
+          "oneOf": [
+            {
+              "type": "object",
+              "description": "Grounding with Bing Search connection configuration",
+              "required": ["type", "options"],
+              "additionalProperties": false,
+              "properties": {
+                "type": { 
+                  "const": "bing_grounding",
+                  "description": "Tool type must be 'bing_grounding'"
+                },
+                "id": { 
+                  "type": ["string", "null"],
+                  "description": "Identifier for Bing Grounding"
+                },
+                "options": {
+                  "type": "object",
+                  "description": "Options for bing grounding tool",
+                  "required": ["tool_connections"],
+                  "additionalProperties": false,
+                  "properties": {
+                    "tool_connections": {
+                      "type": "array",
+                      "description": "List of tool connection strings",
+                      "items": { "type": "string" },
+                      "minItems": 1,
+                      "maxItems": 1
+                    }
+                  }
+                }
+              }
+            },
+            {
+              "type": "object",
+              "description": "Code Interpreter tool configuration. Code interpreter is a tool that can read and interpret information from datasets, generate code, and create graphs and charts.",
+              "properties": {
+                "type": { 
+                  "const": "code_interpreter",
+                  "description": "Tool type must be 'code_interpreter'"
+                },
+                "id": { 
+                  "type": ["string", "null"],
+                  "description": "Identifier for the code interpreter tool"
+                },
+                "options": {
+                  "type": "object",
+                  "description": "Options for code_interpreter tool",
+                  "properties": {
+                    "file_ids": {
+                      "type": "array",
+                      "description": "List of file id used by the code interpreter",
+                      "items": { "type": "string" },
+                      "minItems": 1,
+                      "maxItems": 20
+                    }
+                  },
+                  "required": ["file_ids"],
+                  "additionalProperties": false
+                }
+              },
+              "required": ["type"],
+              "additionalProperties": false
+            },
+            {
+              "type": "object",
+              "description": "OpenAPI tool configuration for weather data",
```

Comment:
"A description of what the openAPI function does, used by the model to choose when and how to call the function."
copied from https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/ai/ai-projects/src/customization/models.ts